authors/apps/authentication/views.py

```diff
@@ -43,10 +49,25 @@ def post(self, request):
         """
         serializer = self.serializer_class(data=user)
         serializer.is_valid(raise_exception=True)
-        serializer.save()
-        user_data = serializer.data
-        user_data['token'] = generate_jwt_token(user['username'])
+        user_data = serializer.validated_data
+
+        # sends email with the activation link with the token
+        subject = 'Authors Haven activation email'
+        message = "Click this link to be activated "
+        current_site = get_current_site(request)
+        token = generate_jwt_token(user['username'])
 
+        activation_link = 'http://' + current_site.domain + '/api/auth/' + token
```

Comment:
@Bileonaire create a check for the current site HTTP protocol and set it instead of explicitly setting 'http'